coral/src/app/features/requests/components/topics/TopicRequests.tsx

```diff
@@ -4,23 +4,29 @@ import { getTopicRequests } from "src/domain/topic/topic-api";
 import { TopicRequestsTable } from "src/app/features/requests/components/topics/components/TopicRequestsTable";
 import { useSearchParams } from "react-router-dom";
 import TopicFilter from "src/app/features/components/table-filters/TopicFilter";
+import { MyRequestFilter } from "src/app/features/components/table-filters/MyRequestFilter";
 
 function TopicRequests() {
   const [searchParams] = useSearchParams();
   const currentTopic = searchParams.get("topic") ?? "";
+  const isMyRequest = searchParams.get("isMyRequest") === "true" ? true : false;
 
   const { data, isLoading, isError, error } = useQuery({
-    queryKey: ["topicRequests", currentTopic],
+    queryKey: ["topicRequests", currentTopic, isMyRequest],
     queryFn: () =>
       getTopicRequests({
         pageNo: "1",
         search: currentTopic,
+        isMyRequest,
       }),
   });
 
   return (
     <TableLayout
-      filters={[<TopicFilter key={"topic"} />]}
+      filters={[
+        <TopicFilter key={"topic"} />,
+        <MyRequestFilter key={"isMyRequest"} />,
```

Comment:
I would add a test in for `TopicRequest` for the switch functionality, too. It will confirm that

- this component uses the `TopicSwitch`
- user using the switch triggers the correct behaviour -> api call with a changed payload. 

Both are the responsibility of this component.

Reply:
Sounds good! 👍🏼